src/main/java/eu/europeana/metis/sandbox/service/dataset/DatasetGeneratorServiceImpl.java

```diff
@@ -48,26 +49,27 @@ private void validateDatasetMetadataAndRecords(DatasetMetadata datasetMetadata,
 
   private Set<Record> processRecordsAndRemoveDuplicates(DatasetMetadata datasetMetadata, List<ByteArrayInputStream> records) {
     return records.stream()
-        .map(ByteArrayInputStream::readAllBytes)
-        .map(recordItem -> getOptionalRecordFromProcessorService(datasetMetadata, recordItem))
-        .flatMap(Optional::stream)
-        .collect(toSet());
+            .map(ByteArrayInputStream::readAllBytes)
+            .map(recordItem -> getOptionalRecordFromProcessorService(datasetMetadata, recordItem))
+            .flatMap(Optional::stream)
+            .collect(toSet());
   }
 
-  private Optional<Record> getOptionalRecordFromProcessorService(DatasetMetadata datasetMetadata, final byte[] recordData) {
+
+  private Optional<Record> getOptionalRecordFromProcessorService(DatasetMetadata datasetMetadata, byte[] record) {
     try {
-      final String recordId = xmlRecordProcessorService.getRecordId(recordData);
+      final RecordEntity recordEntity = recordRepository.save(new RecordEntity(null, null, datasetMetadata.getDatasetId()));
+      final Long recordId = recordEntity.getId();
       return Optional.of(Record.builder()
                                .recordId(recordId)
-                               .europeanaId(EuropeanaIdCreator.constructEuropeanaIdString(recordId, datasetMetadata.getDatasetId()))
                                .datasetId(datasetMetadata.getDatasetId())
                                .datasetName(datasetMetadata.getDatasetName())
                                .country(datasetMetadata.getCountry())
                                .language(datasetMetadata.getLanguage())
-                               .content(recordData)
+                               .content(record)
                                .build());
     } catch (IllegalArgumentException | RecordParsingException processorServiceException) {
```

Comment:
I prefer to keep these exceptions as they are. If we change to a general `Exception`, it means all types of exceptions are captured here and it will log the same message without doing anything else, meanwhile this is not always true for every exception and probably every exception shouldn't be handled the same way. 
Long story short, if this were to catch every `Exception`, this could hide some potential big issues and we don't want that. If you disagree in any level, we can have a chat about this :)

Reply:
I'll change then to `RuntimeException` since all of these are a type of runtime exception :)